spec/tunes/app_version_spec.rb

```diff
@@ -150,10 +171,308 @@
+      end
+
+      it "modifies the watch app data after update" do
+        version.upload_watch_icon!("path_to_jpg")
+        expect(version.watch_app_icon.url).to eq(nil)
+        expect(version.watch_app_icon.original_file_name).to eq("icon1024.jpg")
+        expect(version.watch_app_icon.asset_token).to eq("Purple7/v4/65/04/4d/65044dae-15b0-a5e0-d021-5aa4162a03a3/pr_source.jpg")
+      end
+
+      it "deletes the watch app data" do
+        version.upload_watch_icon!(nil)
+        expect(version.watch_app_icon.url).to eq(nil)
+        expect(version.watch_app_icon.original_file_name).to eq(nil)
+        expect(version.watch_app_icon.asset_token).to eq(nil)
+      end
+    end
+
+    describe "Modifying the app trailers", :trailers do
+      let(:ipad_trailer_path) { "path_to_trailer.mov" }
+      let(:ipad_trailer_preview_path) { "path_to_trailer_preview.jpg" }
+      let(:ipad_external_valid_trailer_preview_path) { "path_to_my_screenshot.jpg" }
+      let(:ipad_external_invalid_trailer_preview_path) { "path_to_my_invalid_screenshot.jpg.jpg" }
+      before do
+        allow(Spaceship::UploadFile).to receive(:from_path) do |path|
+          r = du_uploadtrailer_correct_mov if path == ipad_trailer_path
+          r = du_uploadtrailer_preview_correct_jpg if path == ipad_trailer_preview_path
+          r = du_uploadtrailer_preview_correct_jpg if path == ipad_external_valid_trailer_preview_path
+          r
+        end
+
+        allow(Spaceship::Utilities).to receive(:grab_video_preview) do |path|
+          r = ipad_trailer_preview_path if path == ipad_trailer_path
+          r
+        end
+
+        allow(Spaceship::Utilities).to receive(:portrait?) do |path|
+          r = true if path == ipad_trailer_preview_path
+          r = true if path == ipad_external_invalid_trailer_preview_path
+          r = true if path == ipad_external_valid_trailer_preview_path
+          r
+        end
+
+        allow(Spaceship::Utilities).to receive(:resolution) do |path|
+          r = [900, 1200] if path == ipad_trailer_path
+          r = [768, 1024] if path == ipad_trailer_preview_path
+          r = [700, 1000] if path == ipad_external_invalid_trailer_preview_path
+          r = [768, 1024] if path == ipad_external_valid_trailer_preview_path
+          r
+        end
+
+        json = JSON.parse(du_read_fixture_file("upload_trailer_response_success.json"))
+        allow(client.du_client).to receive(:upload_trailer).and_return(json)
+
+        json = JSON.parse(du_read_upload_trailer_preview_response_success)
+        allow(client.du_client).to receive(:upload_trailer_preview).and_return(json)
+      end
+
+      def trailers(device)
+        version.trailers["English"].find_all { |s| s.device_type == device }
+      end
+
+      def ipad_trailers
+        trailers("ipad")
+      end
+
+      it "cannot add a trailer to iphone35" do
+        expect do
+          version.upload_trailer!(ipad_trailer_path, "English", 'iphone35')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.